src/ts/components/timelinemarkershandler.ts

```diff
@@ -80,61 +80,118 @@ export class TimelineMarkersHandler {
+
+        if (matchingMarker) {
+          matchingMarker.position = markerPosition;
+          matchingMarker.duration = markerDuration;
+
+          this.updateMarkerDOM(matchingMarker);
+        } else {
+          const newMarker: SeekBarMarker = { marker, position: markerPosition, duration: markerDuration };
+          this.timelineMarkers.push(newMarker);
+
+          this.createMarkerDOM(newMarker);
+        }
       }
     });
-
-    // Populate the timeline with the markers
-    this.updateMarkersDOM();
   }
 
-  private updateMarkersDOM(): void {
-    // TODO: For live streams we are removing all markers, and re-adding them.
-    // we should track markers by some internal IDs and just update their positions
-    // This would give us ability to make their pos updating smooth with css transition
-    this.markersContainer.empty();
-
+  private getMarkerCssProperties(marker: SeekBarMarker): { [propertyName: string]: string } {
     const seekBarWidthPx = this.getSeekBarWidth();
 
-    this.timelineMarkers.forEach(marker => {
-      const markerClasses = ['seekbar-marker'].concat(marker.marker.cssClasses || [])
-        .map(cssClass => this.prefixCss(cssClass));
+    const positionInPx = (seekBarWidthPx / 100) * (marker.position < 0 ? 0 : marker.position);
+    const cssProperties: { [propertyName: string]: string } = {
+      '-ms-transform': `translateX(${positionInPx}px)`,
+      '-webkit-transform': `translateX(${positionInPx}px)`,
+      'transform': `translateX(${positionInPx}px)`,
+    };
 
-      const cssProperties: { [propertyName: string]: string } = {
-        'width': marker.position + '%',
-      };
+    if (marker.duration > 0) {
+      const markerWidthPx = Math.round(seekBarWidthPx / 100 * marker.duration);
+      cssProperties['width'] = `${markerWidthPx}px`;
+    }
 
-      if (marker.duration > 0) {
-        const markerWidthPx = Math.round(seekBarWidthPx / 100 * marker.duration);
-        cssProperties['border-right-width'] = markerWidthPx + 'px';
-        cssProperties['margin-left'] = '0';
-      }
+    return cssProperties;
+  }
+
+  private updateMarkerDOM(marker: SeekBarMarker): void {
+    marker.element.css(this.getMarkerCssProperties(marker));
+  }
 
-      this.markersContainer.append(new DOM('div', {
+  private createMarkerDOM(marker: SeekBarMarker): void {
+      const markerClasses = ['seekbar-marker'].concat(marker.marker.cssClasses || [])
+        .map(cssClass => this.prefixCss(cssClass));
+
+      const markerElement = new DOM('div', {
         'class': markerClasses.join(' '),
         'data-marker-time': String(marker.marker.time),
         'data-marker-title': String(marker.marker.title),
-      }).css(cssProperties));
+      }).css(this.getMarkerCssProperties(marker));
+
+      if (marker.marker.imageUrl) {
+        const removeImage = () => {
+          const toRemove = imageElement.get()[0];
+          toRemove.parentElement.removeChild(toRemove);
```

Comment:
```suggestion
          imageElement.remove();
```

Reply:
Utilized in https://github.com/bitmovin/bitmovin-player-ui/pull/318/commits/20c0c1735d663dcae6b550abc75262f18166d1ca